### PR TITLE
Mvvmc managers

### PR DESCRIPTION
**Pull request by ylatuya**

### CODE REVIEW

File: LongoMatch.Core/Store/Templates/SportsTeam.cs [hunk 1]
```diff
@@ -249,6 +249,16 @@ public static SportsTeam DefaultTemplate (int playersCount)
 			return defaultTemplate;
 		}
 
+
+		protected override void ForwardPropertyChanged (object sender, System.ComponentModel.PropertyChangedEventArgs e)
```
Comment (vguzmanp) [inline LongoMatch.Core/Store/Templates/SportsTeam.cs]: using `System.ComponentModel`

File: LongoMatch.GUI/Gui/Panel/SportsTemplatesPanel.cs [hunk 2]
```diff
@@ -325,6 +323,12 @@ void HandleViewModelChanged (object sender, PropertyChangedEventArgs e)
 			}	
 		}
 
+		void HandleBackClicked (object sender, EventArgs e) {
+			viewModel.Save (true);
+			if (BackEvent != null)
+				BackEvent ();
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/SportsTemplatesPanel.cs]: Add braces

File: LongoMatch.GUI/Gui/Panel/SportsTemplatesPanel.cs [hunk 3]
```diff
@@ -325,6 +323,12 @@ void HandleViewModelChanged (object sender, PropertyChangedEventArgs e)
 			}	
 		}
 
+		void HandleBackClicked (object sender, EventArgs e) {
+			viewModel.Save (true);
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/SportsTemplatesPanel.cs]: Here Save changed from false to true, is that correct?

File: LongoMatch.GUI/Gui/Panel/TeamsTemplatesPanel.cs [hunk 4]
```diff
@@ -16,45 +16,41 @@
...
-using Constants = LongoMatch.Core.Common.Constants;
+using VAS.Core.MVVMC;
 using Helpers = VAS.UI.Helpers;
-using Image = VAS.Core.Common.Image;
 
 namespace LongoMatch.Gui.Panel
 {
 	[System.ComponentModel.ToolboxItem (true)]
+	[ViewAttribute ("TeamsManager")]
 	public partial class TeamsTemplatesPanel : Gtk.Bin, IPanel
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/TeamsTemplatesPanel.cs]: This class should be called TeamsManagerPanel, right?

File: LongoMatch.GUI/Gui/Panel/TeamsTemplatesPanel.cs [hunk 5]
```diff
@@ -16,45 +16,41 @@
...
-		List<SportsTeam> teams;
+		TeamsManagerVM viewModel;
 
 		public TeamsTemplatesPanel ()
 		{
 			this.Build ();
-			provider = App.Current.TeamTemplatesProvider;
+
+			panelheader1.ApplyVisible = false;
+			panelheader1.Title = Catalog.GetString ("TEAMS MANAGER");
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/TeamsTemplatesPanel.cs]: This should use the `Title` property of the Panel

File: LongoMatch.GUI/Gui/Panel/TeamsTemplatesPanel.cs [hunk 6]
```diff
@@ -255,283 +253,91 @@ void HandleLeftPlayerButton (object sender, EventArgs e)
...
-			}
-			deleteteambutton.Visible = selected != null;
-			teamtemplateeditor1.Visible = selected != null;
-			if (selected != null) {
-				LoadTeam (selected, iter);
-			}
+		void HandleBackClicked (object sender, EventArgs e) {
+			viewModel.Save (true);
+			if (BackEvent != null)
+				BackEvent ();
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/TeamsTemplatesPanel.cs]: add braces

File: LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs [hunk 7]
```diff
@@ -118,11 +102,42 @@ protected override void OnDestroyed ()
...
+		public string Title {
+			get {
+				return Catalog.GetString ("New project");
+			}
+		}
+
+		public SportsProjectVM ViewModel {
+			set {
+				viewModel = value;
+				project = viewModel.Model;
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs]: Add a FIXME comment to note that this is temporal until `LoadTeams` uses the ViewModel structure

Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs]: In addition, these changes are unrelated to the commit message: it's implementing MVVMC in the `NewProjectPanel`, not just changing the navigation

File: LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs [hunk 8]
```diff
@@ -118,11 +102,42 @@ protected override void OnDestroyed ()
...
+				LoadTeams (project);
+				if (project == null) {
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs]: If project == null, it's already called LoadTeams.

It should go in the else

File: LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs [hunk 9]
```diff
@@ -596,9 +604,7 @@ void HandleQuit (QuitApplicationEvent e)
 		{
 			// When the application is quitting while we are on the new project panel we need to properly destroy widgets.
 			// To do that we go back to the welcome panel, a little bit like the analysis window closes the opened project first.
-			if (BackEvent != null) {
-				BackEvent ();
-			}
+			App.Current.StateController.MoveBack ();
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs]: According to the comment, shouldn't we go to the home, instead of back?

File: LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs [hunk 10]
```diff
@@ -607,9 +613,7 @@ void HandleBackClicked (object sender, EventArgs e)
 				projectperiods1.Pause ();
 			}
 			if (notebook1.Page == firstPage) {
-				if (BackEvent != null) {
-					BackEvent ();
-				}
+				App.Current.StateController.MoveBack ();
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs]: It would be nice to eventually move the _"notebooks to be able to have a back button"_ hack to the states model

File: LongoMatch.GUI/Gui/Panel/OpenProjectPanel.cs [hunk 11]
```diff
@@ -46,17 +48,31 @@ public OpenProjectPanel ()
...
+		{
+			Destroy ();
+			base.Dispose ();
+		}
+
+		public SportsProjectsManagerVM ViewModel {
 			set {
-				projectlistwidget.Fill (value);
+				viewModel = value;
+				projectlistwidget.Fill (viewModel.Model.ToList ());
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/OpenProjectPanel.cs]: Add FIXME

File: LongoMatch.GUI/Gui/Panel/ProjectsManagerPanel.cs [hunk 12]
```diff
@@ -347,17 +353,11 @@ void HandleDeleteClicked (object sender, EventArgs e)
...
-				
+
 			deletedProjects = new List<ProjectLongoMatch> ();
 			foreach (ProjectLongoMatch selectedProject in selectedProjects) {
-				if (openedProject != null && openedProject.ID == selectedProject.ID) {
-					Helpers.MessagesHelpers.WarningMessage (this,
-						Catalog.GetString ("This Project is actually in use.") + "\n" +
-						Catalog.GetString ("Close it first to allow its removal from the database"));
-					continue;
-				}
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Panel/ProjectsManagerPanel.cs]: Where did this behaviour go?

File: LongoMatch/LongoMatch.csproj [hunk 13]
```diff
@@ -23,6 +23,8 @@
         <Variable name="LGM_DEBUG" value="3" />
         <Variable name="LGM_UNINSTALLED" value="1" />
         <Variable name="GTK2_RC_FILES" value="" />
+        <Variable name="MONO_ENV_OPTIONS" value="--arch=64" />
```
Comment (vguzmanp) [inline LongoMatch/LongoMatch.csproj]: Wasn't this causing trouble somewhere?

File: Tests/Services/TestToolsManager.cs [hunk 14]
```diff
@@ -158,17 +158,17 @@ public void TestImportProjectThatNeedsEdition ()
...
 			importer.ImportFunction = () => p;
 			importer.NeedsEdition = true;
-			App.Current.EventsBroker.Publish<ImportProjectEvent> (new ImportProjectEvent ());
-			dbMock.Verify (db => db.Store<ProjectLongoMatch> (p, true), Times.Never ());
+			App.Current.EventsBroker.Publish (new ImportProjectEvent ());
+			dbMock.Verify (db => db.Store (p, true), Times.Never ());
 			Assert.IsTrue (openned);
 
-			App.Current.EventsBroker.Unsubscribe<NewProjectEvent> (et);
+			App.Current.EventsBroker.Unsubscribe<NavigationEvent> (et);
```
Comment (vguzmanp) [inline Tests/Services/TestToolsManager.cs]: This commit ( 42393d3153f991d0e7b6ce3548b3f6c2238f1bcf ) is too big, with too many unrelated changes

File: LongoMatch.GUI/Gui/Panel/SportsTemplatesPanel.cs (hunk 3 above)
Comment (ylatuya) [inline LongoMatch.GUI/Gui/Panel/SportsTemplatesPanel.cs]: Correct, it should be false

File: LongoMatch.GUI/Gui/Panel/SportsTemplatesPanel.cs (hunk 2 above)
Comment (ylatuya) [inline LongoMatch.GUI/Gui/Panel/SportsTemplatesPanel.cs]: And BackEvent should be removed completely since it's no longer part of the IPanel interface

File: LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs (hunk 7 above)
Comment (ylatuya) [inline LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs]: Nope, it's not implementing MVVMC yet, there is no specific ViewModel or Controller, it's just adding a new ViewModel property needed for navigation

File: LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs (hunk 8 above)
Comment (ylatuya) [inline LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs]: I don't understand this comment, LoadTeams needs to be always called.

File: LongoMatch.GUI/Gui/Panel/ProjectsManagerPanel.cs (hunk 12 above)
Comment (vmartos) [inline LongoMatch.GUI/Gui/Panel/ProjectsManagerPanel.cs]: ?

File: LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs (hunk 9 above)
Comment (ylatuya) [inline LongoMatch.GUI/Gui/Panel/NewProjectPanel.cs]: The comment is for the old code, where back always returned to home. We just need to ensure that things are disposed correctly, which is done poping the state from the stack.

File: LongoMatch.GUI/Gui/Panel/ProjectsManagerPanel.cs (hunk 12 above)
Comment (ylatuya) [inline LongoMatch.GUI/Gui/Panel/ProjectsManagerPanel.cs]: This behaviour is not allowed from a long time longomatch, there is no way to go to the projects manager without closing projects first.
